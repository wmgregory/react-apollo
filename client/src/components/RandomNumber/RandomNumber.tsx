import React from 'react';

import { useQuery } from '@apollo/react-hooks';
import { gql } from 'apollo-boost';

const RAND = gql`
  {
    rand
  }
`;

export const RandomNumber: React.FC = () => {
  const { loading, error, data } = useQuery(RAND);

  if (loading) return <p>Loading...</p>;
  if (error) return <p>Error :(</p>;

  const { rand } = data;

  return (
    <div key={rand}>
      <p>Your random number: {rand}</p>
    </div>
  );
};
