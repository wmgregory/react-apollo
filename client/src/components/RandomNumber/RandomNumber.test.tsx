import React from 'react';
import { QueryResult } from '@apollo/react-common';
import * as hooks from '@apollo/react-hooks';

import { shallow } from 'enzyme';

import { RandomNumber } from './RandomNumber';

describe('RandomNumber', () => {
  const mockQuery = { loading: false, error: false, data: { rand: 2 } };

  beforeEach(() => {
    jest.spyOn(hooks, 'useQuery').mockReturnValue((mockQuery as unknown) as QueryResult);
  });
  it('should render with random number', () => {
    const wrapper = shallow(<RandomNumber />);

    expect(wrapper).toBeDefined();
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
