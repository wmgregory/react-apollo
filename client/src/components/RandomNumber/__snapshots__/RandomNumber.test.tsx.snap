// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomNumber should render with random number 1`] = `
"<div>
  <p>
    Your random number: 
    2
  </p>
</div>"
`;
